src/store/albums.js

```diff
@@ -77,6 +77,29 @@ const mutations = {
 		albumNames.forEach(albumName => delete state.albumsFiles[albumName])
 	},
 
+	/**
+	 * Add files to an album.
+	 *
+	 * @param {object} state vuex state
+	 * @param {object} data destructuring object
+	 * @param {string} data.albumName the album id
+	 * @param {string[]} data.fileIds list of files
+	 */
+	setAlbumFiles(state, { albumName, fileIds }) {
+		const albumFiles = state.albumsFiles[albumName] || []
+		state.albumsFiles = {
+			...state.albumsFiles,
+			[albumName]: [
+				...albumFiles,
+				...fileIds.filter(fileId => !albumFiles.includes(fileId)), // Filter to prevent duplicate fileId.
```

Comment:
We should really prefer using Set for performances reasons :)
https://jsben.ch/JNnpx

Reply:
![image](https://user-images.githubusercontent.com/14975046/196747996-f181200e-3d26-4bcf-803f-64c667f8c0d9.png)

Reply:
Can we address that later during a refactoring ?

Reply:
But you made your point :)

Reply:
I actually already made the change in `collectionStoreFactory` that I want to use for albums and shareAlbums

https://github.com/nextcloud/photos/blob/master/src/store/collectionStoreFactory.js#L76-L83

Reply:
But truth for truth, I am actually thinking of more generic store structure instead of having dedicated stores with `collectionStoreFactory`.

Reply:
But first, make it work, then optimize

Reply:
> Can we address that later during a refactoring ?

Absolutely :tada:

Reply:
I'm wondering if there isn't some tools out there that can detect such things :thinking: 
That seems not too crazy. Easy recommendations tbh :see_no_evil: